src/App.tsx

```diff
@@ -1,61 +1,134 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { TodoWithUser } from './types/TodoWithUser';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const getTodoId = (todos: TodoWithUser[]) => {
+  const id = Math.max(...todos.map(todo => todo.id));
+
+  return id + 1;
+};
+
+const todos: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [errorUser, setErrorUser] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [availableTodos, setAvailableTodos] = useState(todos);
+
+  const clearState = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleTitle = (value: string) => {
+    setErrorTitle(false);
+    setTitle(value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimedTitle = title.trim();
+
+    setErrorUser(!userId);
+    setErrorTitle(!trimedTitle);
+
+    if (!trimedTitle || !userId) {
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: getTodoId(availableTodos),
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
+    };
+
+    setAvailableTodos(currentTodos => [...currentTodos, newTodo]);
+
+    clearState();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        method="POST"
+        onSubmit={handleSubmit}
+      >
+
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            <span>Title: </span>
+
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={(event) => (
+                handleTitle(event.target.value)
+              )}
```

Comment:
I recommend to pass just an event to handleTitle function. Than in this function you can get event.target.value. So the syntax will be so after these changes.
```suggestion
              onChange={handleTitle}
```